src/org/labkey/test/stress/AbstractScenario.java

```diff
@@ -0,0 +1,253 @@
+package org.labkey.test.stress;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.remoteapi.CommandException;
+import org.labkey.remoteapi.Connection;
+import org.labkey.test.util.TestDateUtils;
+import org.labkey.test.util.TestLogger;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * Scenario: A set of coordinated simulations across multiple servers, simulating a particular usage event (e.g. one server with a heavy workload of importing large amounts of data while other servers perform simple read-only activities).
+ * This will manage a group of simulations. {@link #startScenario()} will begin the scenario by running all provided simulations in parallel. The method will block briefly to allow simulations to collect some baseline performance data. Then the caller should perform other operations that are expected to impact the performance of the background simulations.<br>
```

Comment:
Nit:
Is it 'baseline performance data' or is it just 'performance data'. (repeated in multiple places)

Reply:
I meant "baseline" to indicate that the data is there for comparison to the perf data that is collected during the stress events (e.g. large sample imports). It is sort of the "zero" for the performance graph.